src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -324,14 +321,6 @@ static IEnumerable<IMethodSymbol> GetToStringMethods(INamedTypeSymbol namedTypeS
                 => namedTypeSymbol.GetMembers("ToString").OfType<IMethodSymbol>().WhereNotNull();
         }
 
-        private static IEnumerable<int> GetIndexesOfParameterType(IMethodSymbol targetMethod, INamedTypeSymbol formatProviderType)
-        {
-            return targetMethod.Parameters
-                .Select((Parameter, Index) => (Parameter, Index))
-                .Where(x => x.Parameter.Type.Equals(formatProviderType))
-                .Select(x => x.Index);
-        }
```

Comment:
I'm now looping over arguments directly and than checking if the parameter is of interest. The new implementation should hopefully be more performant and more straightforward.